src/ui/LinkedText.cpp

```diff
@@ -41,7 +41,7 @@ LinkedText::LinkedText(QObject *parent)
     : QObject(parent)
 {
     // Select things that look like URLs of some kind and allow QUrl::fromUserInput to validate them
-    linkRegex = QRegularExpression(QStringLiteral("([a-z]{3,9}:|www\\.)([^\\s,.);!>]|[,.);!>](?!\\s|$))+"), QRegularExpression::CaseInsensitiveOption);
+    linkRegex = QRegularExpression(QStringLiteral("([a-z]{3,9}:|www\\.)([\\p{L}\\p{N}\\-\\._~:=&/?#]+)"), QRegularExpression::CaseInsensitiveOption);
```

Comment:
Hmm, we are losing a (useful?) trick here.

This section would capture all non-whitespace characters, excluding the characters `,.);!>` _when they are at the end of the URL_. The idea is that if you type `I use http://ricochet.im/. It's neat.`, that period doesn't become part of the link.

Is there a way we can duplicate that effect here, by excluding those characters from the first class and adding them to a second one with the negative forward lookahead?